Source/Api/Controllers/AuthController.cs

```diff
@@ -183,20 +196,27 @@ public class AuthController : ExceptionlessApiController {
                 }
             }
 
+	        if (Settings.Current.EnableActiveDirectoryAuth && !IsAdLoginValid(model.Email, model.Password)) {
+				_logger.Error().Message("Signup failed (bad AD login) for \"{0}\"", model.Email).Tag("Signup").Identity(model.Email).SetActionContext(ActionContext).Write();
+		        return BadRequest("Username and password combination failed to authenticate with Active Directory.");
+	        }
+
             user = new User {
                 IsActive = true,
                 FullName = model.Name,
                 EmailAddress = model.Email,
-                IsEmailAddressVerified = false
+                IsEmailAddressVerified = Settings.Current.EnableActiveDirectoryAuth
             };
             user.CreateVerifyEmailAddressToken();
             user.Roles.Add(AuthorizationRoles.Client);
             user.Roles.Add(AuthorizationRoles.User);
             await AddGlobalAdminRoleIfFirstUserAsync(user);
 
-            user.Salt = Core.Extensions.StringExtensions.GetRandomString(16);
-            user.Password = model.Password.ToSaltedHash(user.Salt);
-
+	        if (!Settings.Current.EnableActiveDirectoryAuth) {
```

Comment:
True, but AD passwords can change over time (some reset every 30 days) and there wouldn't be a way to sync them back to Exceptionless. You could have a case where someone has to remember the password they used X password resets ago in order to change their password or login after switching off AD.